src/org/ods/services/OpenShiftService.groovy

```diff
@@ -133,16 +133,14 @@ class OpenShiftService {
       ]}"""
     ]
 
-    // add build args - TODO: Ensure we are not replacing all buildArgs!!
+    // Add build args
     def buildArgsItems = []
     for (def key : buildArgs.keySet()) {
       def val = buildArgs[key]
       buildArgsItems << """{"name": "${key}", "value": "${val}"}"""
     }
     if (buildArgsItems.size() > 0) {
-      def buildArgsPatch = """{"op": "replace", "path": "/spec/strategy/dockerStrategy", "value": {"buildArgs": [
-      ${buildArgsItems.join(",")}
-    ]}}"""
+      def buildArgsPatch = """{"op": "replace", "path": "/spec/strategy/dockerStrategy/buildArgs", "value": [${buildArgsItems.join(",")}]}"""
```

Comment:
Will this work on an empty dockerStrategy as the buildArgs would not exist yet?

Would we want to also support a way to merge additional parameters (perhaps in another PR) so that one could define some args in the build config and then add unrelated args during the build?

Reply:
Yes, `replace` is equal to `remove` and `add`, and it also works if it does not exist yet.

Your other question is valid too, but a little more complicated. If you feel this to be important, please open a follow-up issue. I'd probably leave it as-is for now ...